db/migrate/20171120000000_add_embed_columns_to_preview_cards.rb

```diff
@@ -0,0 +1,28 @@
+require Rails.root.join('lib', 'mastodon', 'migration_helpers')
+
+class AddEmbedColumnsToPreviewCards < ActiveRecord::Migration[5.1]
+  include Mastodon::MigrationHelpers
+
+  disable_ddl_transaction!
+
+  def change
+    safety_assured do
+      add_column_with_default :preview_cards, :embed_url, :string, default: '', allow_null: false
+
+      reversible do |dir|
+        dir.up do
+          rename_column_concurrently :preview_cards, :html, :embed_html
+          rename_column_concurrently :preview_cards, :width, :embed_width
+          rename_column_concurrently :preview_cards, :height, :embed_height
+        end
+
+        dir.down do
+          cleanup_concurrent_column_rename :preview_cards, :height, :embed_height
+          cleanup_concurrent_column_rename :preview_cards, :width, :embed_width
+          cleanup_concurrent_column_rename :preview_cards, :html, :embed_html
+          execute 'UPDATE preview_cards SET url=embed_url WHERE embed_url IS NOT NULL'
```

Comment:
Since embed_url is not null, it seems that all urls will be changed. I think that should be limited to photo.